### PR TITLE
added: default for StartThreadOpt, THREAD_MEMBER_UPDATE; changed: comment for ThreadMembersUpdate

### PR DESCRIPTION
**Pull request by S3NP41-v**
default instance for StartThreadOpt
event for [ThreadMemberUpdate](https://discord.com/developers/docs/topics/gateway-events#thread-member-update)
changed the comment for ThreadMembersUpdate [according to the docs](https://discord.com/developers/docs/topics/gateway-events#thread-members-update)

### CODE REVIEW

File: src/Discord/Internal/Types/Events.hs [hunk 1]
```diff
@@ -41,11 +41,13 @@ data Event =
   | ThreadCreate               Channel
   -- | thread was updated
   | ThreadUpdate               Channel
+  -- | thread member for the current user was updated
+  | ThreadMemberUpdate         GuildId
```
Comment (L0neGamer) [inline src/Discord/Internal/Types/Events.hs:45]: Counter to what I said, I'm pretty sure this event gives a thread member structure with a guild id as an additional field; would it be possible to give the thread member and the guild id? additionally, are you sure that this event parses? I feel like you don't parse out the `guild_id` sufficiently.

File: src/Discord/Internal/Rest/Channel.hs [hunk 2]
```diff
@@ -308,6 +308,12 @@ instance ToJSON StartThreadOpts where
       , "rate_limit_per_user" .=? startThreadRateLimit
       ]
 
+instance Default StartThreadOpts where
+  def = StartThreadOpts { startThreadName        = ""
```
Comment (L0neGamer) [inline src/Discord/Internal/Rest/Channel.hs:312]: reading the docs it looks like the thread name must be at least one character; maybe we can add that to the above documentation, as well as set a default thread name of `no thread name` or similar to make sure the default works?

Comment (S3NP41-v) [reply src/Discord/Internal/Rest/Channel.hs:312]: maybe the name of `New Thread`? As discord suggests by default
obraz

File: src/Discord/Internal/Types/Events.hs (hunk 1 above)
Comment (S3NP41-v) [reply src/Discord/Internal/Types/Events.hs:45]: should i create a ThreadMemberUpdate record that is the thread member with a guild id, also being honest i am not very sure if it parses

Comment (L0neGamer) [reply src/Discord/Internal/Types/Events.hs:45]: `gateway - received exception [Error in $: Error in $: parsing Snowflake failed, expected String, but encountered Object] while decoding {"t":"THREAD_MEMBER_UPDATE","s":13,"op":0,"d":{"user_id":"?","muted":false,"mute_config":null,"join_timestamp":"2023-12-08T18:40:00.424615+00:00","id":"?","guild_id":"?","flags":0}}` yeah it doesn't parse currently

**Comment by L0neGamer** [comment]
Just want to check that a new feature works, and then I can approve!

File: src/Discord/Internal/Rest/Guild.hs [hunk 3]
```diff
@@ -53,6 +53,8 @@ data GuildRequest a where
   --   'MANAGE_CHANNELS' permission. Returns a list of all of the guild's 'Channel'
   --   objects on success. Fires multiple Channel Update 'Event's.
   ModifyGuildChannelPositions      :: GuildId -> [(ChannelId,Int)] -> GuildRequest [Channel]
+  -- | Returns a list of active threads in a guild
+  ListActiveGuildThreads   :: GuildId -> GuildRequest ListActiveThreads
```
Comment (L0neGamer) [inline src/Discord/Internal/Rest/Guild.hs:57]: Have you tested this works and parses what discord gives us?

Comment (S3NP41-v) [reply src/Discord/Internal/Rest/Guild.hs:57]: i have tested it, but please do so also

Comment (S3NP41-v) [reply src/Discord/Internal/Rest/Guild.hs:57]: yeah went back to look and noticed that i did not add an export for `ListActiveGuildThreads`
maybe `ListActiveThreads` should be moved to `Internal.Types.Guild` ?

Comment (S3NP41-v) [reply src/Discord/Internal/Rest/Guild.hs:57]: > i have tested it, but please do so also

adding an image of a parsed return
obraz

Comment (S3NP41-v) [reply src/Discord/Internal/Rest/Guild.hs:57]: > yeah went back to look and noticed that i did not add an export for `ListActiveThreads` maybe it should be moved to `Internal.Types.Guild` ?

i think that would be best so im going to do that

Comment (S3NP41-v) [reply src/Discord/Internal/Rest/Guild.hs:57]: > yeah went back to look and noticed that i did not add an export for `ListActiveThreads` maybe it should be moved to `Internal.Types.Guild` ?

correction, not `Internal.Types.Guild` but `Internal.Types.Channel`

Comment (L0neGamer) [reply src/Discord/Internal/Rest/Guild.hs:57]: I disagree with the movement; the List Active Threads structure is only ever used in the output of this one thing, so it should be defined and exported from this file as well; it's a request type.

Comment (S3NP41-v) [reply src/Discord/Internal/Rest/Guild.hs:57]: reverted changes and added an export in `Internal.Rest.Guild`

**Review by L0neGamer — APPROVED** [review]
looks good!

**Pull request merged**